### PR TITLE
update our pipeline to use xcode 13.3.1

### PR DESCRIPTION
**Pull request by harrieshin**
### Platforms Impacted
- [x] iOS
- [x] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Update our pipeline to use xcode 13.3.1

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: .ado/azure-pipelines.yml [hunk 1]
```diff
@@ -77,7 +77,7 @@ jobs:
   - job: ApplePR
     displayName: Apple PR
     pool:
-      vmImage: 'macos-11'
+      vmImage: 'internal-macos12'
```
Comment (Saadnajmi) [inline .ado/azure-pipelines.yml:80]: What is this "internal" image? And does it by chance have SIP disabled? :D

**Review by Saadnajmi — APPROVED** [review]

File: .ado/azure-pipelines.yml (hunk 1 above)
Comment (harrieshin) [reply .ado/azure-pipelines.yml:80]: what is SIP?

Comment (Saadnajmi) [reply .ado/azure-pipelines.yml:80]: [SA](https://developer.apple.com/documentation/security/disabling_and_enabling_system_integrity_protection?language=objc)

It prevents us from running automation tests on macOS, and is not disabled on public ADO pools

Comment (harrieshin) [reply .ado/azure-pipelines.yml:80]: but are there actually automation test that are running right now? until we actually have the latest version of xcode in the public image we kind of have to use the internal.

**Review by mischreiber — APPROVED** [review]

**Pull request merged**